README.md

```diff
@@ -1,21 +1,332 @@
+    write(*,*)'output not associated'
+    stop
+endif
+
+! Bind the input tensor
+raw_data_ptr = c_loc(raw_data)
+input_tensors(1) = TF_NewTensor( TF_FLOAT, input_dims, 2, raw_data_ptr, int(128, kind=c_size_t) )
+
+! Run inference
+call TF_SessionRun( session, input_tfoutput, input_tensors, 1, output_tfoutput, output_tensors, 1, &
+    target_opers, 0, stat )
+if (TF_GetCode( stat ) .ne. TF_OK) then
+    call TF_Message( stat, vers )
+    write(*,*) TF_GetCode( stat ), vers
+    call abort
+endif
+
+! Bind output tensor
+call c_f_pointer( TF_TensorData( output_tensors(1)), output_data_ptr, shape(output_data_ptr) )
+write(*,*)'output data', output_data_ptr(1)
+
+if ((output_data_ptr(1) - -0.479371) .gt. 1e-6) then
+    write(*,*)'Output does not match, FAILED!'
+else
+    write(*,*)'Output is correct, SUCCESS!'
+endif
+
+
+! Clean up
+call TF_DeleteTensor( input_tensors(1) )
+call TF_DeleteTensor( output_tensors(1) )
+call TF_DeleteGraph( graph )
+call TF_DeleteSession( session, stat )
+call TF_DeleteStatus( stat )
+
+end program test_program
 ```
-git remote set-url origin git@github.com:Cambridge-ICCS/fortran-tf-lib.git
+#### Generating code with `process_model`
+The example code above illustrates a problem with the TensorFlow C API
+that our Fortran wrapper cannot fix.  To load a model, the library requires
+that the caller knows certain rather opaque model parameters beforehand.
+Often, the values in the example above will work for the `tags` parameter
+to `TF_LoadSessionFromSavedModel`.  However, the values needed for
+`TF_GraphOperationByName` (in this case `serving_default_input_1`, etc)
+are more likely to be different.
+
+To address this, we provide a Python script, `process_model` that will
+read a Keras SavedModel and output a simple Fortran module intended to
+provide a base for the user to start from.  The appropriate values will
+be read from the model and hard-coded into the Fortran code.
+
+E.g.
 ```
-or
+process_model -o fortran_code.f90  my_model
 ```
-git remote set-url origin https://github.com/Cambridge-ICCS/fortran-tf-lib.git
+
+### 3. Build the code
+
+The code now needs to be compiled and linked against our installed library.
+
+#### CMake
+If our project were using cmake we would need the following in the
+`CMakeLists.txt` file to find the the tf-lib installation and link it to the
+executable.
+
+This can be done by adding the following to the `CMakeLists.txt` file:
+```
+find_package(FortranTensorFlow)
+target_link_libraries( <executable> PRIVATE FortranTensorFlow::fortran-tf )
+message(STATUS "Building with Fortran TensorFlow coupling")
+```
+and using the `-DCMAKE_PREFIX=</path/to/fortran-tf-libs/lib64/cmake>` flag when running cmake.
+
+When running the generated code you may also need to add the location of the
+`.so` files to your `LD_LIBRARY_PATH` unless installing in a default location:
+```
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:<path/to/fortran-tf-libs>/lib64
```

Comment:
Is this preferable to using a `-L` linker flag?
I'll update the pytorch docs to bring in line or at least mention both.

Reply:
This is different.
Simon has removed `make` instructions as keeps it simpler, and if someone is using `make` they are probably know how to link a library already (right....??)